libraries/rush-lib/src/cli/RushPnpmCommandLineParser.ts

```diff
@@ -421,8 +421,13 @@ export class RushPnpmCommandLineParser {
           keepEnvironment: true
         },
         onStdoutStreamChunk,
-        (exitCode: number, signal: string) => {
-          process.exitCode = exitCode;
+        (exitCode: number | null, signal: NodeJS.Signals | null) => {
+          if (typeof exitCode === 'number') {
+            process.exitCode = exitCode;
+          } else {
+            // Terminated by a signal
+            process.exitCode = 1;
```

Comment:
Semantically `143` is closer to being the expected error code for signal termination, at least on Linux. Error code 143 is used when the OS kills the process.

Reply:
On Linux the exit code is 128 plus the signal number, so 143 would correspond to `SIGKILL` but not `SIGTERM` for example. On Windows the exit code uses a more complex formula.

But in this scenario the process being terminated by a signal is `pnpm` not `rush-pnpm`, so I'm not sure it's appropriate for `rush-pnpm` to try to reproduce the exit code of its child as that kind of misrepresents what happened. (?)

Maybe we should keep it simple.